traits/trait_list_object.py

```diff
@@ -285,8 +285,7 @@ def __setitem__(self, key, value):
                 pass
         super().__setitem__(key, value)
 
-        if added != removed:
-            self.notify(normalized_index, removed, added)
+        self.notify(normalized_index, removed, added)
```

Comment:
So we still want to suppress the notification in the case that there's no _structural_ change to the list - i.e., no items were either removed or added. The other methods do this consistently - e.g., clearing an empty list doesn't notify.

IOW, I think we want
```
if added or removed:
    self.notify(...)
```

Reply:
(Or perhaps `if len(added) > 0 or len(removed) > 0`, or whatever style best matches the similar checks elsewhere.)

To clarify, something like `myobj.mylist[0:0] = []` shouldn't notify, but if `myobj.mylist[0:3]` happens to be equal to `[1, 2, 3]` already, then `myobj.mylist[0:3] = [1, 2, 3]` *should* still notify.

Reply:
Done. Update PR description to reflect the change as well